### ISSUE TITLE
[CI/CD] Migrate prowjobs out of Neighbor's Prow - Jellyfish

### ISSUE THREAD

**Issue by KacperMalachowski** [kyma-project/test-infra#9495]
**Description**
The prow jobs owned by the jellyfish team have to be removed and if necessary substituted with something else.
To simplify the process here is a list of prow jobs that should be migrated by jellyfish team.

  - [x] pull-main-reconciler-integration-k3d - will be dropped later
  - [x] pre-main-reconciler-unit-test - will be dropped later
  - [x] pre-main-reconciler-lint - will be dropped later
  - [x] pre-main-reconciler-publish-pr-cli - will be dropped later
  - [x] pull-template-op-lint - move to GH Action in the Template Operator repository
  - [x] pull-template-op-tests - move to GH Action in the Template Operator repository
  - [x] pre-kyma-cli - will be dropped later
  - [x] pull-cli-unit-test - will be dropped later
  - [x] pre-main-control-plane-reconciler-e2e - will be dropped later
  - [x] pre-main-control-plane-reconciler-e2e-latest-release - will be dropped later
  - [x] pull-control-plane-rec-upgrade-kyma2-stable-to-main-grdnr - will be dropped later
  - [ ] release-template-operator-assets - move to GH Action in the Template Operator repository
  - [x] ~post-main-unstable-cli~ - build job
  - [x] post-kyma-cli - will be dropped later
  - [x] ~rel-kyma-cli~ - build job
  - [x] reconciler-control-plane-image-bump - will be dropped later
  - [x] nightly-main-reconciler - will be dropped later
  - [x] nightly-main-reconciler-e2e - will be dropped later
  - [x] kyma-upgrade-gardener-kyma2-to-main-reconciler-main - will be dropped later

  - [x] pre-main-hydroform - will be dropped later
  - [x] post-main-hydroform - will be dropped later

**Comment by janmedrek** [comment]
In the first iteration, we will focus on all the pipelines that are not related to the CLI or Hydroform

**Comment by janmedrek** [comment]
We need to move the following pipelines to the Template Operator repository
- pull-template-op-lint
- pull-template-op-tests
- release-template-operator-assets

Those should be converted/rewritten to the GH Actions.

**Comment by janmedrek** [comment]
As for the rest of the pipelines, I will prepare a follow-up and/or create a new ticket.

**Comment by janmedrek** [comment]
https://github.com/kyma-project/test-infra/issues/9873

**Comment by nesmabadr** [comment]
The `release-template-operator-assets` is still pending on the neighbors to give the right access to the GCP project Kyma-project to upload the assets
